File: php/features.php
<?php
declare(strict_types=1);

// Fruityvice SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FruityviceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FruityviceBaseFeature();
            case "test":
                return new FruityviceTestFeature();
            default:
                return new FruityviceBaseFeature();
        }
    }
}

File: py/features.py
# Fruityvice SDK feature factory

from feature.base_feature import FruityviceBaseFeature
from feature.test_feature import FruityviceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FruityviceBaseFeature(),
        "test": lambda: FruityviceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
